src/alarms/alarmtemplates/TemplateForm.tsx

```diff
@@ -0,0 +1,346 @@
+import { FormattedMessage } from 'react-intl';
+import formStyles from './../../styles/Forms.module.css';
+import buttonStyles from './../../styles/Buttons.module.css';
+import templateIcon from "../../images/templates@3x.svg";
+
+interface Props {
+  currentTemplate?: any
+};
+interface PropsFromDispatch {
+  addNotification: (message: string | number, timeout: number) => void
+};
+
+export const availableParameters = [
+  {
+    parameter: "first_name",
+    parameterText: "[[var:first_name]]",
+    label: "first name recipient",
+    description: "First name of the recipient"
+  },
+  {
+    parameter: "last_name",
+    parameterText: "[[var:last_name]]",
+    label: "last name recipient",
+    description: "Last name of the recipient"
+  },
+  {
+    parameter: "email",
+    parameterText: "[[var:email]]",
+    label: "email recipient",
+    description: "E-mail address of the recipient",
+    templateType: "email"
+  },
+  {
+    parameter: "alarm_name",
+    parameterText: "[[var:alarm_name]]",
+    label: "name alarm",
+    description: "Name of the alarm"
+  },
+  {
+    parameter: "warning_value",
+    parameterText: "[[var:warning_value]]",
+    label: "value threshold",
+    description: "Numerical value of the threshold"
+  },
+  {
+    parameter: "warning_level",
+    parameterText: "[[var:warning_level]]",
+    label: "status threshold",
+    description: "Name of the threshold"
+  },
+  {
+    parameter: "threshold_timestamp",
+    parameterText: "[[var:threshold_timestamp]]",
+    label: "timestamp threshold",
+    description: "Time the threshold was crossed"
+  },
+  {
+    parameter: "trigger_timestamp",
+    parameterText: "[[var:trigger_timestamp]]",
+    label: "trigger timestamp",
+    description: "Moment the alarm was analysed"
+  },
+  {
+    parameter: "warning_timestamp",
+    parameterText: "[[var:warning_timestamp]]",
+    label: "warning timestamp",
+    description: "Time the peak or trough is reached"
+  },
+  {
+    parameter: "first_timestamp",
+    parameterText: "[[var:first_timestamp]]",
+    label: "first timestamp",
+    description: "First time the alarm was triggered"
+  }
+];
+
+const TemplateForm: React.FC<Props & PropsFromDispatch & RouteComponentProps> = (props) => {
+  const { currentTemplate } = props;
+  const selectedOrganisation = useSelector(getSelectedOrganisation);
+
+  const initialValues = currentTemplate ? {
+    name: currentTemplate.name,
+    type: convertToSelectObject(currentTemplate.type, currentTemplate.type.toUpperCase()),
+    subject: currentTemplate.subject,
+    message: currentTemplate.type === 'sms' ? currentTemplate.text : currentTemplate.html // if email, read html field
+  } : {
+    name: null,
+    type: convertToSelectObject('email', 'EMAIL'),
+    subject: null,
+    message: ''
+  };
+
+  const onSubmit = (values: Values) => {
+    const body = {
+      name: values.name,
+      subject: values.subject,
+      text: values.message,
+      html: values.message
+    };
```

Comment:
here I still need to update both `text` and `html` field because the `text` field is required by backend. If we only send `html` when creating a email template then it will response with a 400 error code because of missing `text` field.